src/bci_build/containercrate.py

```diff
@@ -0,0 +1,40 @@
+"""Crate to handle multibuild containers in the generator."""
+
+
+class ContainerCrate:
+    """ContainerCrate is combining multiple container build flavors.
+
+    This provides package-central functions like generating _service and
+    _multibuild files.
+    """
+
+    def __init__(self, containers: list):
+        """Assign the crate for every container."""
+        self._all_build_flavors: dict[tuple, set] = {}
```

Comment:
The crate is to unify over `(os_version, container.package_name)`. if I understand correctly, you're unifying over all given containers (which could be many different versions, which have different build flavors depending on version and also have different package names, like in the tomcat case). However, we do not want to collect flavors that are not built for this version or not build for this package name. hence, we need to group them by this, which is not working as far as I can see in your proposal. Maybe I'm missing something.

Reply:
As an example, with tomcat, there are two flavors on SP6 (17, 21) and 3 on tumbleweed (17, 21, 22). the generated _multibuild file needs to be specific to the os-version. in theory it could even be specific to the package name. 

dict/set are the most efficient ways to aggregate it.